src/hooks/boardwalk/useWallet.tsx

```diff
@@ -167,9 +186,107 @@ const useWallet = () => {
          addToast(msg, 'error');
       }
    };
+
+   /** Generate an invoice from mintless wallet or active cashu wallet */
+   const generateInvoice = useCallback(
+      async (amount: number): Promise<{ invoice: string; checkingId: string }> => {
+         let result: { invoice: string; checkingId: string };
+         if (isMintless) {
+            /* TODO: need some way to check status of invoice */
+            result = {
+               invoice: await mintlessReceive(amount),
+               checkingId: '',
+            };
+         } else {
+            if (!activeWallet) {
+               throw new Error('Active wallet not set and mintless receive is not enabled');
+            }
+            const quote = await activeWallet.createMintQuote(amount);
+            result = {
+               invoice: quote.request,
+               checkingId: quote.quote,
+            };
+
+            /* save mint quote until we claim proofs */
+            addPendingMintQuote({ ...quote, amount, keysetId: activeWallet.keys.id });
```

Comment:
why do we need this stored to local storage? isn't it enough to have pending transaction stored in the history?